flink-table-store-connector/src/main/java/org/apache/flink/table/store/connector/sink/StoreSinkWriter.java

```diff
@@ -94,8 +94,9 @@ private RecordWriter getWriter(BinaryRowData partition, int bucket) {
                 k ->
                         overwrite
                                 ? fileStoreWrite.createEmptyWriter(
-                                        partition, bucket, compactExecutor)
-                                : fileStoreWrite.createWriter(partition, bucket, compactExecutor));
+                                        partition.copy(), bucket, compactExecutor)
```

Comment:
Is this an existing bug in the previous version, right ?

Reply:
> Is this an existing bug in the previous version, right ?

Yes, at that moment, FLINK-27652 is still not merged; the formal PR will rebase the latest master branch.